AppCenterAnalytics/AppCenterAnalyticsTests/MSDoubleTypedPropertyTests.m

```diff
@@ -0,0 +1,72 @@
+#import "MSDoubleTypedProperty.h"
+#import "MSConstants+Internal.h"
+#import "MSTestFrameworks.h"
+
+@interface MSDoubleTypedPropertyTests : XCTestCase
+
+@end
+
+@implementation MSDoubleTypedPropertyTests
+
+- (void)testNSCodingSerializationAndDeserializationWorks {
+
+  // If
+  MSDoubleTypedProperty *sut = [MSDoubleTypedProperty new];
+  sut.type = @"type";
+  sut.name = @"name";
+  sut.value = 12.23432;
+
+  // When
+  NSData *serializedProperty = [NSKeyedArchiver archivedDataWithRootObject:sut];
+  MSDoubleTypedProperty *actual = [NSKeyedUnarchiver unarchiveObjectWithData:serializedProperty];
+
+  // Then
+  XCTAssertNotNil(actual);
+  XCTAssertTrue([actual isKindOfClass:[MSDoubleTypedProperty class]]);
+  XCTAssertEqualObjects(actual.name, sut.name);
+  XCTAssertEqualObjects(actual.type, sut.type);
+  XCTAssertEqual(actual.value, sut.value);
+}
+
+- (void)testSerializeToDictionaryWorks {
+
+  // If
+  MSDoubleTypedProperty *sut = [MSDoubleTypedProperty new];
+  sut.name =  @"propertyName";
+  sut.value = 0.123;
```

Comment:
Is there any reason we're using different values for different test cases?  It's a bad pattern to use.  If we want to use multiple values we should put them in a list and iterate over them for each test.